drivers/ieee802154/ieee802154_nrf5.c

```diff
@@ -988,7 +988,7 @@ static int nrf5_configure(const struct device *dev,
 
 #if defined(CONFIG_IEEE802154_CSL_ENDPOINT)
 	case IEEE802154_CONFIG_CSL_RX_TIME:
-		nrf5_data.csl_rx_time = config->csl_rx_time;
+		nrf5_data.csl_rx_time = target_time_convert_to_64_bits(config->csl_rx_time);
```

Comment:
Wondering if we could get rid of `nrf5_data.csl_rx_time` and directly call to `nrf_802154_csl_writer_anchor_time_set` here.

Reply:
The value is stored in`static uint64_t  m_csl_anchor_time`, so IMO yes, we could call `nrf_802154_csl_writer_anchor_time_set` only on change of `csl_rx_time` here.

Reply:
I think that's a nice optimization. We could easily get rid of `nrf5_schedule_rx` and `nrf5_config_csl_period` static functions too then.

Reply:
removed `nrf5_receive_at`, `nrf5_config_csl_period` and  `nrf5_schedule_rx`